### ISSUE TITLE
Landing on the wrong screen after unlocking is done

### ISSUE THREAD

**Issue by hffvld** [brave/brave-ios#7657]
### Description: 

Landing on the wrong screen after unlocking is done

### Preconditions
- Make sure device unlock is enabled. You can enable `PIN` option.
- `Sync` is already enabled

### Steps to Reproduce 
  1. Launch Brave > Tree-dot menu > Settings
  2. Sync > Observe `PIN` screen to unlock the device
  3. Tap `Cancel` > Tap `Unlock` again
  4. Enter correct PIN > Observe

### Actual result: 

- The user is landing on `Settings` screen instead of `Sync` screen if canceling the first attempt to unlock the device and then trying to unlock it again. 
- The same issue was observed for `Logins & Password` screen where the user landed on a previous screen.

https://github.com/brave/brave-ios/assets/128532965/24c635f6-b507-469d-8301-55f527540c9f

### Expected result:

- The user must land on `Sync` screen when unlocking the device by going to `Sync` settings.
- The user must be able to `copy` or `reveal` the saved password when unlocking the device by going to `Saved passwords` settings.

### Reproduces how often: [Easily reproduced, Intermittent Issue]

Easily reproduced

### Brave Version: 

Brave build: 1.52 (23.6.28.13)
BraveCore: 1.52.129 (114.0.5735.198)

- Can you reproduce this issue with the most recent build from TestFlight? Yes
- Can you reproduce this issue with the previous version of the current build from TestFlight? Yes
- Can you reproduce this issue with the current build from AppStore? NA

### Device details: 

iPhone 14 [iOS 16.5]

### Website problems only:
- did you check with Brave Shields down? 
- did you check in Safari/Firefox (WkWebView-based browsers)?

**Comment by hffvld** [comment]
I also filed this issue #7656 where the user can't go back to the previous screen from `Unlock` screen if tapping `Cancel` button. So, maybe if the fix for #7656 will be to `close Unlock screen and return to the previous screen` then it should also fix this issue as well since the second attempt to enter PIN will be unavailable.

**Comment by hffvld** [comment]
Verified on `iPhone 14` using version(s):
```
Device/OS: iPhone 14 [iOS 16.5.1]
Brave build: 1.56 (23.7.18.21)
BraveCore: 1.56.6 (115.0.5790.90)
```
---------------------------

**STEPS:**
  1. Launch Brave > Tree-dot menu > Settings
  2. Sync > Observe `PIN` screen to unlock the device
  3. Tap `Cancel` > Tap `Unlock` again
  4. Enter correct PIN > Observe

**ACTUAL RESULTS:**
- Verified that a user is landing on the correct `Sync settings` screen

---------------------------

https://github.com/brave/brave-ios/assets/128532965/83981fe5-d05a-4588-b768-be25cb393ef4
